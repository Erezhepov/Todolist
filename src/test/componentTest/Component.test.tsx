import Component from "./Component";
import {fireEvent, render, screen} from "@testing-library/react";
import '@testing-library/jest-dom'
import userEvent from "@testing-library/user-event";


describe('Component', () => {
    test('render Component', () => {
        render(<Component />)
        const btn: any = screen.queryByRole('button')
        expect(screen.queryByText(/title/i)).toBeNull()
        fireEvent.click(btn)
        expect(screen.queryByText(/title/i)).toBeInTheDocument()
        fireEvent.click(btn)
        expect(screen.queryByText(/title/i)).toBeNull()
    })

    test('render input', () => {
        render(<Component />)
        const input = screen.getByPlaceholderText('enter')
        const value = screen.queryByTestId('input-value')
        expect(value).toContainHTML('')
        userEvent.type(input, '123')
        expect(value).toBeInTheDocument()
        expect(value).toContainHTML('123')
    })
})