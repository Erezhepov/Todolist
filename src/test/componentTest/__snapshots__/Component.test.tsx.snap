// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component render Component 1`] = `
<input
  placeholder="lol"
  type="text"
  value=""
/>
`;
